From 422a1638d3e137693ac5935fec997733c41f1f41 Mon Sep 17 00:00:00 2001
From: Dentrax <furkan.turkal@chainguard.dev>
Date: Sun, 31 Aug 2025 22:39:11 +0300
Subject: [PATCH] Fix entitlements for cacerts

When running with Java default trust stores, access to both cacerts
and jssecacerts might be required. Grant such access to avoid warnings
in the error logs.

Signed-off-by: Dentrax <furkan.turkal@chainguard.dev>
Co-authored-by: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
---
 .../initialization/HardcodedEntitlements.java        | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/HardcodedEntitlements.java b/libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/HardcodedEntitlements.java
index 53e70f04..33d9d2cd 100644
--- a/libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/HardcodedEntitlements.java
+++ b/libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/HardcodedEntitlements.java
@@ -169,6 +169,9 @@ class HardcodedEntitlements {
             Path trustStorePath = trustStore != null
                 ? Path.of(trustStore)
                 : Path.of(System.getProperty("java.home")).resolve("lib/security/jssecacerts");
+            // Always add below two
+            Path jssecacertsPath = Path.of(System.getProperty("java.home")).resolve("lib/security/jssecacerts");
+            Path cacertsPath = Path.of(System.getProperty("java.home")).resolve("lib/security/cacerts");
 
             Collections.addAll(
                 serverScopes,
@@ -184,7 +187,14 @@ class HardcodedEntitlements {
                     "org.bouncycastle.fips.core",
                     // read to lib dir is required for checksum validation
                     List.of(
-                        new FilesEntitlement(List.of(FilesEntitlement.FileData.ofBaseDirPath(LIB, READ))),
+                        new FilesEntitlement(
+                            List.of(
+                                FilesEntitlement.FileData.ofBaseDirPath(LIB, READ),
+                                FilesEntitlement.FileData.ofPath(trustStorePath, READ),
+                                FilesEntitlement.FileData.ofPath(jssecacertsPath, READ),
+                                FilesEntitlement.FileData.ofPath(cacertsPath, READ)
+                            )
+                        ),
                         new ManageThreadsEntitlement()
                     )
                 )
-- 
2.39.5 (Apple Git-154)

